packages/playwright-ct-react/registerSource.mjs

```diff
@@ -17,14 +17,16 @@
 // @ts-check
 // This file is injected into the registry as text, no dependencies are allowed.
 
-import React from 'react';
-import ReactDOM from 'react-dom';
+import * as React from 'react';
+import { createRoot } from 'react-dom/client';
 
 /** @typedef {import('../playwright-test/types/component').Component} Component */
 /** @typedef {import('react').FunctionComponent} FrameworkComponent */
 
 /** @type {Map<string, FrameworkComponent>} */
 const registry = new Map();
+/** @type {WeakMap<Element, import('react-dom/client').Root>} */
+const rootRegistry = new WeakMap();
```

Comment:
How do you utilize the weakness of this map? It seems like every time root element is removed w/o harness knowing, this is a hard bug rather than a leak.

Reply:
Done